WalletWasabi/Crypto/ZeroKnowledge/Challenge.cs

```diff
@@ -0,0 +1,42 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public static class Challenge
+	{
+		public static Scalar Build(GroupElement publicPoint, GroupElement nonce, IEnumerable<GroupElement> generators)
```

Comment:
i don't think it's the statement per se, but a variable appearing in that statement, "public point" makes sense to me because it's a public input

Reply:
Actually the full statement consists of the "public point" and all the generator(s) (multiple generators in case of the knowledge of representation proof system).

Reply:
fair enough, same disagreement as the call the other day ;-)

to me the statement is an static expression with free variables for the public inputs and constant terms for the generators, and obviously the actual statement needs the public inputs bound... if the public inputs are committed along with the challenges, everything else is always the same so that just needs an identifier such as the name of the proof, and the generators can be inferred from that.

committing the generators separately is superfluous since they are completely static, and comitting a new statement each time with the public inputs plugged into that statement seems unnecessarily complicated, but all are equivalent

Reply:
So I should leave it `publicPoint`?